### PR TITLE
Switch to Noble for Rolling. (#193)

### PR DESCRIPTION
**Pull request by wep21**
* backport #193 into jazzy

### CODE REVIEW

**Comment by christophebedard** [comment]
I would also just wait until the `jazzy` branch freeze is over until merging this.

**Review by christophebedard — APPROVED** [review]
Alright, the freeze has been lifted, and this only modifies local GitHub CI, so no need to run ROS 2 CI. Thanks for the PR.

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -13,13 +13,19 @@ jobs:
...
-        colcon-mixin-repository: https://raw.githubusercontent.com/colcon/colcon-mixin-repository/master/index.yaml
+        colcon-defaults: |
+          {
+            "build": {
+              "mixin": [
+                "coverage-gcc"
+              ]
+            }
+          }
         target-ros2-distro: rolling
```
Comment (christophebedard) [inline .github/workflows/test.yml]: Oh wait. I realized that you backported the commit from #193. Can you change `rolling` to `jazzy` here?

File: .github/workflows/test.yml [hunk 2]
```diff
@@ -37,12 +43,18 @@ jobs:
...
+        colcon-defaults: |
+          {
+            "build": {
+              "mixin": [
+                "asan-gcc"
+              ]
+            }
+          }
         package-name: libstatistics_collector
         target-ros2-distro: rolling
```
Comment (christophebedard) [inline .github/workflows/test.yml]: Can you change `rolling` to `jazzy` here too?

**Review by christophebedard — CHANGES_REQUESTED** [review]
Just need to change `rolling` to `jazzy` in a couple places.

File: .github/workflows/test.yml (hunk 1 above)
Comment (wep21) [inline .github/workflows/test.yml]:
```suggestion
        target-ros2-distro: jazzy
```

File: .github/workflows/test.yml (hunk 2 above)
Comment (wep21) [inline .github/workflows/test.yml]:
```suggestion
        target-ros2-distro: jazzy
```

File: .github/workflows/test.yml (hunk 1 above)
Comment (wep21) [reply .github/workflows/test.yml]: Thank you for your suggestion. I changed the distro.

**Review by christophebedard — APPROVED** [review]

**Pull request merged**